std/math.d

```diff
@@ -1764,14 +1772,22 @@ real cosh(real x) @safe pure nothrow @nogc
     //  cosh = (exp(x)+exp(-x))/2.
     // The naive implementation works correctly.
     const real y = exp(x);
-    return (y + 1.0/y) * 0.5;
+    return (y + 1.0/y) * 0.5L;
 }
 
 /// ditto
-double cosh(double x) @safe pure nothrow @nogc { return cosh(cast(real) x); }
+double cosh(double x) @safe pure nothrow @nogc
+{
+    const double y = exp(x);
+    return (y + 1.0/y) * 0.5;
+}
 
 /// ditto
-float cosh(float x) @safe pure nothrow @nogc  { return cosh(cast(real) x); }
+float cosh(float x) @safe pure nothrow @nogc
+{
+    const float y = exp(x);
+    return (y + 1.0/y) * 0.5f;
```

Comment:
Division, addition and multiplication all in double precision because of the `1.0`.

Reply:
Fixed.